projects/solar/src/test_data_tx.c

```diff
@@ -0,0 +1,93 @@
+#include "can.h"
+#include "can_msg_defs.h"
+#include "can_transmit.h"
+#include "data_store.h"
+#include "data_tx.h"
+#include "event_queue.h"
+#include "gpio.h"
+#include "interrupt.h"
+#include "log.h"
+#include "ms_test_helpers.h"
+#include "soft_timer.h"
+#include "solar_events.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+int count = -1;                 // count of can msgs sent, incremented in rx_handler
+bool changeable_is_set = true;  // value that tells data_tx if data points are set
+static CanMessage s_can_msg;    // variable used to hold can msgs and verify values
+
+bool TEST_MOCK(data_store_get_is_set)(DataPoint data_point, bool *is_set) {
+  return changeable_is_set;
+}
+
+StatusCode TEST_MOCK(data_store_get)(DataPoint data_point, uint32_t *value) {
+  *value = data_point;
+  return STATUS_CODE_OK;
+}
```

Comment:
I'd prefer if you didn't mock the `data_store` functions and instead used the actual data store in the test. That way we keep mocks to a minimum and the test is less flimsy (e.g. the current implementation of `data_store_get_is_set` won't work because you have to assign to `is_set` and the function should return a `StatusCode`). Then to set all the data points you can just loop through and set them all (you may want a helper function for doing this).

Also for future reference, if you use mocks in a test you have to add the test to the project's `rules.mk` file, there are examples in `solar/rules.mk`.